MigrationGuide-2.1.md

```diff
@@ -0,0 +1,93 @@
+3. **NEW**: Log file message format for `keter` logs is now  
+   `${time}|${module}:${line_num}|${log_level}> ${msg}`  
+  
+   **OLD**: Log file message format for `keter` logs used to be  
+   `%Y-%m-%d %H:%M:%S.${%q_9} UTC: ${msg}`
+
+Where:
+- `${time}` is time formatted as `%Y%m%d_%H%M%S.${%q_2}`.
+- `${module}` is the `keter` source module the message was logged from.
+- `${line_num}` is the source line number within `${module}` the message was logged from.
+- `${log_level}` is the log level (`Info` | `Warn` | `Error` | `Other`).
+- `${msg}` is the log message itself.
+- `${%q_N}` (where N is 9 or 2 above) is the same as `%q` (picosecond formatting) from the `time` package, but truncated to the first N digits.
+
+## What are some examples of the aforementioned changes?
+
+In the same order:
+
+1. (See below)
+2. **NEW**:  
+    * `log/keter.log`
+    * `log/keter.log.1`
+    * `log/keter.log.2`
+    * `log/app-blah.log`
+    * `log/app-blah.log.1`
+    * `log/app-blah.log.2`
+ 
+   **OLD**:  
+    * `log/keter/current.log`
+    * `log/keter/2023-12-31_245603.log`
+    * `log/keter/2023-01-01_010203.log`
+    * `log/app-blah/current.log`
+    * `log/app-blah/2023-12-31_245603.log`
+    * `log/app-blah/2023-01-01_010203.log`
+3. **NEW**:
+    * `2023-03-11 14:37:17.06|Keter.Main:84|Info> Launching initial`
+    * `2023-04-12 09:02:07.92|Keter.Whatever:279|Other> Something something`
+   
+   **OLD**:
+    * `2023-03-11 14:37:17.069101123 UTC: Launching initial`
+    * `2023-04-12 09:02:07.921023056 UTC: Something something`
+
+## What suggestions do you have for addressing these breaking changes?
+
+In the same order:
+
+1. Update the directories you are reading the log files from appropriately.
+2. Update how you are referencing the actual log file name(s) apppropriately.
+     * If you had previously depended on the timestamp in the name of rotated log files, consider parsing the log message timestamps _within_ the individual log files instead. The timestamp of the first log message in any file should accurately tell you at what time logging was ***rotated to*** that file. You can then subtract the number suffixed to its file name by 1 to then find the name of the log file that was ***rotated out*** at that time, if maintaing old semantics is important to you.
+3. Update parsing of log messages appropriately.
+
+## Other features you may want to make use of
+
+Stderr logging has now been added! Please consider using this option instead if you are, say, integrating `keter` with `systemd`, which captures stderr output for you. By default, without updating your `keter` config, `keter` will still log to rotating files as usual. As such, this is not a breaking change.
+
+You can enable stderr logging by setting `rotate-logs: false` in your `keter` config.
```

Comment:
we probably want to add a not on the stderr option in the readme as well for systemd based distributions: https://github.com/snoyberg/keter#configuring-startup

Reply:
Added~